src/lang/modifyAst/boolean.ts

```diff
@@ -167,6 +167,26 @@ export function findAllChildrenAndOrderByPlaceInCode(
   artifact: Artifact,
   artifactGraph: ArtifactGraph
 ): Artifact[] {
+  const pathToCompositeSolidMap: { [key: string]: string[] } = {}
+  for (const [id, artifact] of artifactGraph) {
+    if (artifact.type === 'compositeSolid') {
+      for (const pathId of artifact.solidIds) {
+        if (pathToCompositeSolidMap[pathId]) {
+          pathToCompositeSolidMap[pathId].push(id)
+        } else {
+          pathToCompositeSolidMap[pathId] = [id]
+        }
+      }
+      for (const pathId of artifact.toolIds) {
+        if (pathToCompositeSolidMap[pathId]) {
+          pathToCompositeSolidMap[pathId].push(id)
+        } else {
+          pathToCompositeSolidMap[pathId] = [id]
+        }
+      }
+    }
+  }
```

Comment:
@jtran I wante dot ask you about this

I'm building up a little map to go from a pathId to compositeSolids if they exist, because the composite solids in the artifactGraph done have edges going both ways, so there's no way to go from an path to it's compositeSolid

I did try adding the edge going the other way in https://github.com/KittyCAD/modeling-app/compare/kurt-composite-attempt?expand=1
But I danno it just didn't work and I couldn't figure out why, I thought maybe I screwed up the merge logic, but it looks okay to me?

But what do you think, solve this in rust, or just keep what I've got?

Reply:
I'll wait for @jtran's input before merging

Reply:
When you say "it just didn't work", what symptoms are you actually seeing? The Mermaid chart generation wasn't updated, but when I did that, the charts show that the edges are bidirectional for the sim tests we have. But for some reason, edges in the graph are lost. Here's the diff of the `intersect_cubes` sim test. The path (3) correctly gains the edge to the composite (17), but it loses the edge to the solid2d (13). Same thing with the other path (4).

<img width="298" alt="Screenshot 2025-05-06 at 11 31 01 AM" src="https://github.com/user-attachments/assets/c5ab8a7e-e1b2-4295-b1ba-79d17dbfc0ca" />

Reply:
Wait, I think I fixed it. The above diff made it obvious. I'll push to your branch.

Reply:
Pushed a few commits. This one is the fix: e23e6d2ef0e857b8d45f7c2b383c5327c181ba80.

Reply:
Thank you!

Looking now.

Reply:
Oh, what a silly mistake :/

Reply:
Okay, I've removed my little mapping, just waiting for CI to be green.